pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt

```diff
@@ -1,9 +1,41 @@
 package com.pubnub.api.endpoints.push
 
+import cocoapods.PubNubSwift.PubNubObjC
 import com.pubnub.api.Endpoint
+import com.pubnub.api.PubNubException
 import com.pubnub.api.models.consumer.push.PNPushAddChannelResult
+import com.pubnub.api.onFailureHandler
+import com.pubnub.api.onSuccessHandler
+import com.pubnub.api.v2.callbacks.Consumer
+import com.pubnub.api.v2.callbacks.Result
+import kotlinx.cinterop.ExperimentalForeignApi
+import platform.Foundation.NSString
+import platform.Foundation.NSUTF8StringEncoding
+import platform.Foundation.dataUsingEncoding
 
 /**
  * @see [PubNub.addPushNotificationsOnChannels]
  */
-actual interface AddChannelsToPush : Endpoint<PNPushAddChannelResult>
\ No newline at end of file
+actual interface AddChannelsToPush : Endpoint<PNPushAddChannelResult>
+
+@OptIn(ExperimentalForeignApi::class)
+open class AddChannelsToPushImpl(
+    private val pubnub: PubNubObjC,
+    private val channels: List<String>,
+    private val deviceId: String
+): AddChannelsToPush {
+    override fun async(callback: Consumer<Result<PNPushAddChannelResult>>) {
+        val deviceIdData = (deviceId as NSString).dataUsingEncoding(NSUTF8StringEncoding)
```

Comment:
do you know if this works?
the documentation says it should be possible to cast String to NSString and vice versa, but the IDE complains that this cast doesn't succeed

it's OK if you don't know yet, we can leave it and when we write tests we'll get a crash an fix it

Reply:
Sure, I'll verify it and let you know